packages/medusa/src/api-v2/admin/inventory-items/helpers.ts

```diff
@@ -0,0 +1,29 @@
+import { MedusaContainer } from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+
+export const refetchInventoryItem = async (
+  inventoryItemId: string,
+  scope: MedusaContainer,
+  fields: string[]
+) => {
+  const remoteQuery = scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "inventory_item",
+    variables: {
+      filters: { id: inventoryItemId },
+      skip: 0,
+      take: 1,
+    },
+    fields: fields,
+  })
+
+  // TODO: Why does the response type change if you pass skip and take, vs not passing it?
+  // Also, why does the data change (in this case, not doing skip and take will not return the lazy fields of stockedQuantity and reserved_quantity)
```

Comment:
This seems strange, did you check if this happens on regular `list` vs `listAndCount` calls?